packages/blade/src/components/Badge/Badge.stories.tsx

```diff
@@ -0,0 +1,191 @@
+const Page = (): ReactElement => {
+  const figmaURL = useMakeFigmaURL([
+    {
+      themeTokenName: 'paymentTheme',
+      lightModeURL:
+        'https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=8110%3A417',
+      darkModeURL:
+        'https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=8110%3A417',
+    },
+    {
+      themeTokenName: 'bankingTheme',
+      lightModeURL:
+        'https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=9727%3A118573',
+      darkModeURL:
+        'https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=9727%3A118573',
+    },
+  ]);
+
+  return (
+    <>
+      <Title />
+      <Subtitle>
+        Badges are used to show small amount of color coded metadata, which are ideal for getting
+        user attention.
+      </Subtitle>
+      <Link withArrow={true} href={figmaURL} target="_blank" rel="noreferrer noopener">
+        View in Figma
+      </Link>
+      <br />
+      <br />
+      <Title>Usage</Title>
+      <Highlight language="tsx">{`import { Badge } from '@razorpay/blade/components'; \nimport type { BadgeProps } from '@razorpay/blade/components';`}</Highlight>
+      <Title>Example</Title>
+      <Subtitle>
+        This is the default badge. You can change the properties of this button using the controls
+        in the table below.
+      </Subtitle>
+      <Primary />
+      <Title>Properties</Title>
+      <ArgsTable story={PRIMARY_STORY} />
+      <Description markdown=">Note: `icon` prop accepts an `IconComponent` of Blade which should be used as:" />
+      <Highlight language="tsx">{`import { Badge, InfoIcon } from '@razorpay/blade/components'; \n\n &ltBadge icon={InfoIcon}>Label&lt/Badge>`}</Highlight>
+      <Stories />
+    </>
+  );
+};
+
+export default {
+  title: 'Components/Badge',
+  component: BadgeComponent,
+  argTypes: {
+    icon: {
+      name: 'icon',
+      type: 'select',
+      options: Object.keys(iconMap),
+    },
+  },
+  parameters: {
+    docs: {
+      page: Page,
+    },
+  },
+} as Meta<BadgeProps>;
+
+const BadgeTemplate: ComponentStory<typeof BadgeComponent> = ({ icon, children, ...args }) => {
+  const IconComponent = iconMap[(icon as unknown) as string];
+  return (
+    <BadgeComponent icon={IconComponent} {...args}>
+      {children}
+    </BadgeComponent>
+  );
+};
+
+export const Badge = BadgeTemplate.bind({});
+Badge.args = {
+  children: 'Label',
+  variant: 'neutral',
+  fontWeight: 'regular',
+  contrast: 'low',
+  size: 'small',
+};
+Badge.storyName = 'Default';
+
+const BadgesWithVariantTemplate: ComponentStory<typeof BadgeComponent> = ({ ...args }) => {
+  const variants: Array<NonNullable<BadgeProps['variant']>> = [
+    'positive',
+    'negative',
+    'notice',
+    'information',
+    'neutral',
+  ];
+
+  const getLabel = (label: string): string => {
+    return args.fontWeight === 'bold' ? label.toUpperCase() : capitalize(label);
+  };
```

Comment:
Should we add this logic in story? Won't people think just by looking at the story that if fontWeight is bold the badge automatically makes it uppercase?

Reply:
They won't. If they see the code below the story, they can see that the label text is capitalized. Anyway, even if they do the moment they use it they'll realise it doesn't work that way so its okay. Not a huge learning curve there

Reply:
Non blocker: personally in favor of keeping storybooks vanilla, without any special logic